config/quality/checkstyle/checkstyle.xml

```diff
@@ -1,238 +1,240 @@
+        <!-- <module name="ImportOrder"/> -->
+        <module name="RedundantImport" />
+        <module name="UnusedImports" />
 
         <!-- Checks for Javadoc comments.                     -->
         <!-- See http://checkstyle.sourceforge.net/config_javadoc.html -->
         <module name="JavadocMethod">
             <property name="scope" value="package" />
-            <property name="allowMissingParamTags" value="true" />
+            <property name="allowMissingPropertyJavadoc" value="true" />
             <property name="allowMissingThrowsTags" value="true" />
-            <property name="allowMissingReturnTag" value="true" />
             <property name="allowThrowsTagsForSubclasses" value="true" />
             <property name="allowUndeclaredRTE" value="true" />
-            <property name="allowMissingPropertyJavadoc" value="true" />
-            <property name="severity" value="ignore" />
         </module>
         <module name="JavadocType">
             <property name="scope" value="package" />
-            <property name="severity" value="ignore" />
         </module>
+        <!-- We ignore this warning, because variable should be self-explanatory, thus eliminating the need for Javadoc -->
         <module name="JavadocVariable">
-            <property name="scope" value="package" />
             <property name="severity" value="ignore" />
         </module>
         <module name="JavadocStyle">
-            <property name="checkEmptyJavadoc" value="true" />
-            <property name="severity" value="ignore" />
+            <property name="checkFirstSentence" value="false" />
         </module>
 
+        <!-- Checks for Metrics. -->
+        <!-- See http://checkstyle.sourceforge.net/config_metrics.html -->
+        <module name="BooleanExpressionComplexity" />
+        <module name="ClassDataAbstractionCoupling" />
+        <module name="ClassFanOutComplexity" />
+        <module name="CyclomaticComplexity" />
+        <module name="JavaNCSS" />
+        <module name="NPathComplexity" />
+
+
+        <!-- Miscellaneous other checks.                   -->
+        <!-- See http://checkstyle.sourceforge.net/config_misc.html -->
+        <module name="ArrayTypeStyle" />
+        <module name="AvoidEscapedUnicodeCharacters" />
+        <module name="CommentsIndentation" />
+        <module name="DescendantToken" />
+        <module name="FileContentsHolder" />
+        <module name="FinalParameters" />
+        <module name="Indentation" />
+        <module name="OuterTypeFilename" />
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+CR\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TodoComment">
+            <property name="format" value="(?i)\s+TODO\s+" />
+            <property name="severity" value="info" />
+        </module>
+        <module name="TrailingComment" />
```

Comment:
I've read it all before commenting  :)

But soemtimes you just need to put a symbolic comment or a very minor just because you want to remind yourself/coworker something you've been doing in the previous session.

I can see myself using this against the proposed style without hurting anybody.